support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -3,14 +3,16 @@ package utils
 import com.gu.i18n.Currency.GBP
 import com.gu.i18n.{Country, CountryGroup, Currency}
 import com.gu.support.redemptions.RedemptionData
-import com.gu.support.workers._
+import com.gu.support.workers.{GiftRecipient, _}
+import com.gu.support.zuora.api.ReaderType
 import com.typesafe.scalalogging.LazyLogging
 import org.joda.time.LocalDate
 import services.stepfunctions.CreateSupportWorkersRequest
+import services.stepfunctions.CreateSupportWorkersRequest.GiftRecipientRequest
 
 object CheckoutValidationRules {
   def validatorFor(productType: ProductType): CreateSupportWorkersRequest => Boolean = productType match {
-    case _: DigitalPack => DigitalPackValidation.passes
+    case d: DigitalPack => DigitalPackValidation.passes(d)
```

Comment:
it would be great to revisit the validation stuff, and consolidate it with the code that turns the data into the State model.
We already know this logic is duplicated between client and server side, but even within the server side it happens either when we convert it into the CreatePaymentMethodState, or later in a leaky way if the types don't retain the information.
In terms of the backend I think we should aim to do this once in one go, (and ideally retain the information in the types, if appropriate to avoid dealing "this can't happen" situations later)  This also means we are less likely to forget to add this validation when we change the model.
background reading
 https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/